src/api/endpoints/redemption_requests.py

```diff
@@ -0,0 +1,365 @@
+"""RedemptionRequest Module."""
+
+from flask import request, g, current_app
+from flask_restplus import Resource
+
+from api.utils.notifications.email_notices import send_email
+from api.utils.auth import token_required, roles_required
+from api.utils.helpers import find_item, paginate_items, response_builder
+from api.utils.helpers import serialize_redmp
+from api.utils.marshmallow_schemas import redemption_request_schema
+from api.utils.marshmallow_schemas import basic_info_schema, redemption_schema
+from ..models import Society, RedemptionRequest, Center
+
+
+class PointRedemptionAPI(Resource):
+    """
+    Resource handling all point redemption requests.
+
+    Only made by society presidents.
+    """
+
+    @classmethod
+    @token_required
+    @roles_required(["society president"])
+    def post(cls):
+        """Create Redemption Request."""
+        payload = request.get_json(silent=True)
+        if not payload:
+            return response_builder(dict(
+                message="Redemption request must have data.",
+                status="fail"
+            ), 400)
+
+        result, errors = redemption_request_schema.load(payload)
+
+        if errors:
+            status_code = redemption_request_schema.context.get(
+                'status_code')
+            validation_status_code = status_code or 400
+            return response_builder(errors, validation_status_code)
+
+        center = Center.query.filter_by(name=result.get('center')).first()
+        # TODO: Before creating a redemption request make sure society has
```

Comment:
This is still not done. Should be a high priority after MVP release.

Reply:
Update: Done now.

Reply:
👍